### PR TITLE
edk2toolext/edk2_logging.py: Log rust errors

### PR DESCRIPTION
**Pull request by Javagedes**
Adds the ability to detect rust build errors and print them at the end of the build when the build fails.

image

### CODE REVIEW

**Comment by Javagedes** [comment]
@makubacki is this what you were requesting in #658? We don't currently try to catch the c-style "clippy-like" explanations when compiles fail. See test `test_scan_compiler_output_gcc_mixed_actual`.

I know on the mu_basecore side of things you want to catch silent failures in the plugin, but not 100% sure what you are looking for on the extensions side of things as described in #658

**Comment by makubacki** [comment]
> @makubacki is this what you were requesting in #658? We don't currently try to catch the c-style "clippy-like" explanations when compiles fail. See test `test_scan_compiler_output_gcc_mixed_actual`.
> 
> I know on the mu_basecore side of things you want to catch silent failures in the plugin, but not 100% sure what you are looking for on the extensions side of things as described in #658

I was looking to add more context in error messages so problems can better be understood. The output shown in the PR description looks fine to me.

File: edk2toolext/edk2_logging.py [hunk 1]
```diff
@@ -231,6 +231,7 @@ def output_compiler_error(match, line, start_txt="Compiler"):
     build_py_error_exp = re.compile(r"error (\d+)E:")
     linker_error_exp = re.compile(r"error LNK(\d+):")
     warning_exp = re.compile(r"warning [A-Z]?(\d+):")
+    rust_error_exp = re.compile(r"^(-->|\d+\s*\||error:)")
```
Comment (makubacki) [inline edk2toolext/edk2_logging.py:234]: Could you add a few tests?

Comment (Javagedes) [reply edk2toolext/edk2_logging.py]: the regex was reduced to only `error:`, `error[errotype]:` and `-->` to reduce conflicts with catching unexpected c code. Example in the PR summary. 

Tests have been added. Thanks!

**Review by makubacki — APPROVED** [review]

File: edk2toolext/edk2_logging.py (hunk 1 above)
Comment (Javagedes) [reply edk2toolext/edk2_logging.py]: I have added tests. Thanks!

**Pull request merged**